data/src/client.rs

```diff
@@ -243,7 +334,72 @@ impl Client {
                 return None;
             }
             _ if batch_tag.is_some() => {
-                let events = self.handle(message, context)?;
+                let events = if let Some(target) = batch_tag
+                    .as_ref()
+                    .and_then(|batch| self.batches.get(batch))
+                    .and_then(|batch| batch.chathistory_target.clone())
+                {
+                    if let Some(ChatHistoryRequest {
+                        subcommand,
+                        message_reference,
+                        ..
+                    }) = self.chathistory_requests.get(&target)
+                    {
```

Comment:
You can collapse these two if let branches together by calling `self.chathistory_requests.get` in an `and_then` and pattern matching the `Some(ChatHistoryRequest` in the root if let